src/main/java/de/ellpeck/rockbottom/api/entity/player/Gamemode.java

```diff
@@ -0,0 +1,16 @@
+package de.ellpeck.rockbottom.api.entity.player;
+
+public enum Gamemode {
```

Comment:
Can we move this out of an Enum and maybe store it in a registry? Allow mods to be able to create their own game modes maybe? Not sure if that is needed or not but it's worth to think about that

Reply:
Gamemodes need to change a lot of game internals and even with a lot of events, it isn't suitable for extending.

Reply:
It's okay as is right now. It will be perfect for 0.4 release. If we think we can add on to it by allows custom game modes then we can change the system in the future.